tests/cupy_tests/test_cudnn.py

```diff
@@ -343,3 +343,115 @@ def test_call(self):
         else:
             with self.assertRaises(self.err):
                 self.call()
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'ksize': [1, 3, 5],
+    'stride': [1, 2, 4],
+    'auto_tune': [True, False],
+    'layout': layouts
+}))
+@unittest.skipUnless(cudnn_enabled, 'cuDNN is not available')
+class TestConvolutionInvalid(unittest.TestCase):
+
+    def setUp(self):
+        self._workspace_size = cudnn.get_max_workspace_size()
+        if libcudnn.getVersion() < 7500:
+            self.skip = True
+            self.skip_msg = "cuDNN version is older than 7.5.0"
+            return
+        else:
+            self.skip = False
+        n = 16
+        x_c, y_c = 64, 64
+        x_h, x_w = 32, 32
+        y_h, y_w = x_h // self.stride, x_w // self.stride
+        self.pad = (self.ksize - 1) // 2
+        if self.layout == libcudnn.CUDNN_TENSOR_NHWC:
+            x_shape = (n, x_h, x_w, x_c)
+            y_shape = (n, y_h, y_w, y_c)
+            W_shape = (y_c, self.ksize, self.ksize, x_c)
+        else:
+            x_shape = (n, x_c, x_h, x_w)
+            y_shape = (n, y_c, y_h, y_w)
+            W_shape = (y_c, x_c, self.ksize, self.ksize)
+        self.x = cupy.ones(x_shape, dtype=self.dtype)
+        self.W = cupy.ones(W_shape, dtype=self.dtype)
+        self.y = cupy.empty(y_shape, dtype=self.dtype)
+        self.gx = cupy.empty(x_shape, dtype=self.dtype)
+        self.gW = cupy.empty(W_shape, dtype=self.dtype)
+        self.gy = cupy.ones(y_shape, dtype=self.dtype)
+        cudnn.set_max_workspace_size(0)
+
+    def tearDown(self):
+        cudnn.set_max_workspace_size(self._workspace_size)
+
+    def call_forward(self):
+        cudnn.convolution_forward(
+            self.x, self.W, None, self.y,
+            pad=(self.pad, self.pad), stride=(self.stride, self.stride),
+            dilation=(1, 1), groups=1,
+            auto_tune=self.auto_tune, tensor_core="always",
+            d_layout=self.layout, w_layout=self.layout)
+
+    def test_forward(self):
+        if self.skip:
+            return unittest.SkipTest(self.skip_msg)
+        self.call_forward()
```

Comment:
Hm, is this testing invalid cases?

Reply:
Actually, no.  Invalid tests for "convolution_foward" should be skipped?

Reply:
I personally feel that it's better to keep the intent of each test clear (test the error handling in this case), so I'd say yes.

Reply:
By the way, it's difficult to produce a failure for forward?

Reply:
I found it difficult as long as "correct" config is used. It is not impossible to produce a failure but the config is really "wrong".

Reply:
If it's difficult, it might be fine, but I'd if possible like to have the errors introduced tested. What would be a typical really "wrong" config?

Reply:
For example, tensor size or data type mismatch between x and y, etc. An error is raised here with these "wrong" config, but the cause is different.

Reply:
Ah, okay I understand it's difficult. I think it's okay to proceed with merging this PR.